EKF/common.h

```diff
@@ -43,6 +43,28 @@
 
 #include <matrix/math.hpp>
 
+namespace ecl{
+	inline float powf(float x, int exp)
+	{
+		float ret;
+		if (exp > 0) {
+			ret = x;
+			for (int count = 1; count < exp; count++) {
+				ret *= x;
+			}
+			return ret;
+		} else if (exp < 0) {
+			ret = x;
```

Comment:
I think you could replace this case with `return 1.0f / ecl::powf(x, -exp);`, and then the `ret` can be simplified above as well into only the `if (exp > 0) ` case.